### PR TITLE
Dynamic Views in LTC with JIT [WIP]

### PR DESCRIPTION
**Pull request by Krovatkin**
Fixes #ISSUE_NUMBER

### CODE REVIEW

File: lazy_tensor_core/lazy_tensor_core/csrc/init_python_bindings.cpp [hunk 1]
```diff
@@ -683,6 +683,63 @@ void InitLtcModuleBindings(py::module m) {
...
+
+    // replace lazy tensors with meta tensors
+    // so we can do shape inference
+    for (auto i: c10::irange(args.size() - 1)) {
+      if (THPVariable_Check(args[i + 1].ptr())) {
+        auto& t = THPVariable_Unpack(args[i + 1].ptr());
+        auto lt = torch::lazy::GetLtcTensor(t);
+        device = lt.GetDevice();
+        tvals.push_back(lt.GetIrValue());
+        auto mt = at::empty_strided(t.sizes(), t.strides(), at::TensorOptions(at::kMeta).dtype(t.scalar_type()));
```
Comment (wconstab) [inline lazy_tensor_core/lazy_tensor_core/csrc/init_python_bindings.cpp:711]: can't just do args[i + 1].to(kMeta)?

File: lazy_tensor_core/lazy_tensor_core/csrc/init_python_bindings.cpp [hunk 2]
```diff
@@ -683,6 +683,63 @@ void InitLtcModuleBindings(py::module m) {
...
+        stack.push_back(mt);
+      } else {
+        ivals.push_back(torch::jit::toTypeInferredIValue(args[i+1]));
```
Comment (wconstab) [inline lazy_tensor_core/lazy_tensor_core/csrc/init_python_bindings.cpp:714]: just to be clear, this case is for scalars?

Comment (Krovatkin) [reply lazy_tensor_core/lazy_tensor_core/csrc/init_python_bindings.cpp:714]: yes

File: lazy_tensor_core/lazy_tensor_core/csrc/init_python_bindings.cpp (hunk 1 above)
Comment (Krovatkin) [reply lazy_tensor_core/lazy_tensor_core/csrc/init_python_bindings.cpp:711]: I could check; my concern was this could trigger materialization?

Comment (wconstab) [reply lazy_tensor_core/lazy_tensor_core/csrc/init_python_bindings.cpp:711]: actually scratch the .to suggestion, appears even that is not needed.

Our LazyNativeFunctions code just does this:

            auto out_meta = at::meta::nll_loss_backward(grad_output, self, target, weight, reduction, ignore_index, total_weight);
            
            where grad_output is the at::tensor that is passed into the nativefunction.  I guess you can just pass the lazytensor directly to meta func and it converts it inside?

**Pull request closed without merging**